### PR TITLE
[Redesign] Asset selector redesign and bug fixes

### PR DESCRIPTION
**Pull request by nikarm22**
Changes:
-  Restructure AssetPicker for UI v2
-  Improve AssetPickerv2 UI
-  Add swap chains button to redesign UI
-  Show all networks as scrollable in Asset Picker
-  Display Token names in asset picker and add Explorer URLs

Screenshots:
Screenshot from 2024-08-14 04-35-29
Screenshot from 2024-08-14 04-35-20
Screenshot from 2024-08-14 04-35-08

### CODE REVIEW

File: wormhole-connect/src/views/v2/Bridge/AssetPicker/ChainList.tsx [hunk 1]
```diff
@@ -136,11 +148,11 @@ const ChainList = (props: Props) => {
...
             .includes(chainSearchQuery.toLowerCase());
         })
-      : topChains;
+      : props.chainList;
 
     return (
-      <List>
-        <ListItem>
+      <List className={classes.searchList}>
+        <ListItem className={classes.searchBar}>
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/AssetPicker/ChainList.tsx]: @nikarm22 Can we leave the search bar outside the scrollable section? The scrollbar should only appear for the list of chains. I've previously added the search bar as part of the list for better alignment, but for this we can move it out.

**Review by emreboga — CHANGES_REQUESTED** [review]
We need to display the scrollbar only for the list of chains, leaving the search bar out.

File: wormhole-connect/src/views/v2/Bridge/SwapChains/index.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  );
+
+  const canSwap =
+    fromChain &&
+    !config.chains[fromChain]?.disabledAsDestination &&
+    toChain &&
+    !config.chains[toChain]?.disabledAsDestination;
+
+  const swap = () => {
+    if (!canSwap) return;
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/SwapChains/index.tsx]: Nitpick: You can combine these two checks in one.

File: wormhole-connect/src/views/v2/Bridge/SwapChains/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    if (isTransactionInProgress) return;
+    dispatch(swapChains());
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/SwapChains/index.tsx:40]: Just to confirm, there won't be any race-condition or UI flickering when calling these in parallel, right?

File: wormhole-connect/src/views/v2/Bridge/AssetPicker/TokenItem.tsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
+import React, { memo } from 'react';
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/AssetPicker/TokenItem.tsx:1]: Is there anything new in this file or mostly split/copy from `AssetPicker/index.tsx`?

File: wormhole-connect/src/views/v2/Bridge/AssetPicker/SearchableList/index.tsx [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  renderFn: (item: T, index: number) => ReactNode;
+  filterFn: (item: T, query: string) => boolean;
+};
+
+function SearchableList<T>(props: SearchableListProps<T>): ReactNode {
+  const { classes } = useStyles();
+  const [query, setQuery] = useState('');
+
+  const filteredList = useMemo(() => {
+    if (!query && props.initialItems) return props.initialItems;
```
Comment (emreboga) [inline wormhole-connect/src/views/v2/Bridge/AssetPicker/SearchableList/index.tsx]: Nitpick: TBH we need to enforce the [curly rule](https://eslint.org/docs/latest/rules/curly) in eslint to have a unified and more readable block statements. For now this is ok, I can add the rule and fix all these at once in a separate PR/

File: wormhole-connect/src/views/v2/Bridge/SwapChains/index.tsx (hunk 3 above)
Comment (nikarm22) [reply wormhole-connect/src/views/v2/Bridge/SwapChains/index.tsx:40]: Both actions are fully sync operations, rendering is done after both are complete.

File: wormhole-connect/src/views/v2/Bridge/AssetPicker/TokenItem.tsx (hunk 4 above)
Comment (nikarm22) [reply wormhole-connect/src/views/v2/Bridge/AssetPicker/TokenItem.tsx:1]: Most of the file is split from `AssetPicker/TokenList.tsx`.

Added/Changed part is added token display name and explorer link.
As per design I'll have the USD value as well.

**Review by emreboga — APPROVED** [review]
LGTM!

**Pull request merged**